src/charts/bar.js

```diff
@@ -683,7 +683,7 @@ define(function(require) {
         function drawHorizontalGridLines() {
             maskGridLines = svg.select('.grid-lines-group')
                 .selectAll('line.vertical-grid-line')
-                .data(xScale.ticks(4))
+                .data(xScale.ticks(xTicks).slice(1))
```

Comment:
Oh I see, you just followed the pattern in `stacked-bar` and `grouped-bar`. Makes sense.

Reply:
Spot on! But there is a point to it - it's removing the first element so that a grid line isn't drawn on top of the axis line (well, the `.extended-y-line`/`.extended-x-line`).

Reply:
Makes sense! Thanks a lot for your contribution!

Reply:
You're welcome, thanks for looking at it so quickly!